mlflow/models/container/__init__.py

```diff
@@ -106,14 +106,18 @@ def _install_pyfunc_deps(
 
     # NB: If we don't use virtualenv or conda env, we don't need to install mlflow here as
     # it's already installed in the container.
-    if len(activate_cmd) and install_mlflow:
-        install_mlflow_cmd = [
-            "pip install /opt/mlflow/."
-            if _container_includes_mlflow_source()
-            else f"pip install mlflow=={MLFLOW_VERSION}"
-        ]
-        if Popen(["bash", "-c", " && ".join(activate_cmd + install_mlflow_cmd)]).wait() != 0:
-            raise Exception("Failed to install mlflow into the model environment.")
+    if len(activate_cmd):
+        if _container_includes_mlflow_source():
+            # If the MLflow source code is copied to the container,
+            # we always need to run `pip install /opt/mlflow` otherwise
+            # the MLflow dependencies are not installed.
+            install_mlflow_cmd = ["pip install /opt/mlflow/."]
+        elif install_mlflow:
+            install_mlflow_cmd = [f"pip install mlflow=={MLFLOW_VERSION}"]
```

Comment:
```suggestion
            install_mlflow_cmd = [f"pip install mlflow=={MLFLOW_VERSION}"]
        else:
            install_mlflow_cmd = None
```

Reply:
Why didn't the tests you ran above catch this? Is it because _container_includes_mlflow_source is always true in our tests 🤔

Reply:
Only the case that we start container, and set env_manager != local , the issue is triggered.